cmd/list/listconsumergroup_test.go

```diff
@@ -0,0 +1,67 @@
+package list
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
+)
+
+type mockConsumerListener struct{ mock.Mock }
+
+func (m *mockConsumerListener) ListConsumerGroups() (map[string]string, error) {
+	args := m.Called()
+	return args.Get(0).(map[string]string), args.Error(1)
+}
+
+func (m *mockConsumerListener) GetConsumerGroupsForTopic(consumerGroups []string, topic string) (chan string, error) {
+	args := m.Called(consumerGroups, topic)
+	return args.Get(0).(chan string), args.Error(1)
+}
+
+func TestListGroupsReturnsSuccess(t *testing.T) {
+	mockConsumer := new(mockConsumerListener)
+	admin := consumerGroupAdmin{mockConsumer}
+	mockChannel := make(chan string, 0)
+
+	consumerGroupsMap := map[string]string{"consumer1": "", "consumer2": ""}
+	mockConsumer.On("ListConsumerGroups").Return(consumerGroupsMap, nil)
+
+	mockConsumer.On("GetConsumerGroupsForTopic", []string{"consumer1", "consumer2"}, "").Return(mockChannel, nil)
+
+	err := admin.ListGroups("")
+
```

Comment:
Since this feature is listing for a topic, we could mandate topic being not empty.

Reply:
let's add logger and assert the result being consumer1, consumer2 (if it's simpler)

Reply:
Have set the topic flag as mandatory.